packages/luigis-box/src/FrontendApi/Resolver/Recommendation/RecommendationQuery.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\LuigisBoxBundle\FrontendApi\Resolver\Recommendation;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+use Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxClient;
+use Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadDataFactory;
+use Shopsys\LuigisBoxBundle\Model\Type\TypeInLuigisBoxEnum;
+use Shopsys\ProductFeed\LuigisBoxBundle\Model\FeedItem\LuigisBoxProductFeedItem;
+
+class RecommendationQuery extends AbstractQuery
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider $productElasticsearchProvider
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxClient $luigisBoxClient
+     * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadDataFactory $luigisBoxBatchLoadDataFactory
+     */
+    public function __construct(
+        protected readonly ProductElasticsearchProvider $productElasticsearchProvider,
+        protected readonly LuigisBoxClient $luigisBoxClient,
+        protected readonly LuigisBoxBatchLoadDataFactory $luigisBoxBatchLoadDataFactory,
+    ) {
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return \Shopsys\FrameworkBundle\Model\Product\Product[]
+     */
+    public function recommendationQuery(Argument $argument): array
+    {
+        $type = $argument['type'];
+        $limit = $argument['limit'];
+        $userIdentifier = $argument['userIdentifier'];
+        $productIds = [];
+
+        if (isset($argument['productUuids'])) {
+            $products = $this->productElasticsearchProvider->getSellableProductArrayByUuids($argument['productUuids']);
```

Comment:
Maybe we could fetch just the ids here? Let's discuss it please, I believe you can find inspiration in https://github.com/shopsys/shopsys/commit/4d2b53a578b2cdf1d44809b58fb42c9aac93f932 (which is sadly in `15.0` only)